mcr_scene_segmentation/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(mcr_scene_segmentation)
 
+# Compile as C++11 for ROS Kinetic and newer
+add_compile_options(-std=c++11
+  -O3
+  -march=native
```

Comment:
O3 is fine. @abhishek098 said there was some issue with -march=native though

Reply:
it was the issue with manipulation, not perception. this has been tested on youbot

Reply:
-march=native has problem with orocos KDL, I think I need to build from source. @mhwasil has tested it on robot and it works.

Reply:
Comparison with and without compiler optimization

```
### With -O3 #####################################################################
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.117249931
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.085714018
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.106139226
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.088469942
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.135032847
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.091846879
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.114429925
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.084608953
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.095011244
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.081217100
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.121488057
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.081694069
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.127270613
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.074411664
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.104175505
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.089835323
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.132596845
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.079628339
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.115722424
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.084952730
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.093805578
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.073107223
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.102415261
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.080962065

### Without -O3 ###################################################################
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.214613904
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.110857691
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.243467654
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.100388221
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.190539251
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.109664021
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.194542185
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.106416152
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.173597675
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.103750205
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.187207892
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.101182464
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.191387076
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.108483901
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.240434523
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.100857923
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.209278441
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.112250207
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.197830559
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.107400438
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.205611622
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.099196049
[ INFO] [/mcr_perception/scene_segmentation]: OCTREE construction time 0.176610696
[ INFO] [/mcr_perception/scene_segmentation]: got norms
[ INFO] [/mcr_perception/scene_segmentation]: Saving pointcloud to /home/robocup/tmp/
[ INFO] [/mcr_perception/scene_segmentation]: SEGMENTATION time 0.120639480
```

Reply:
Every time we compile code locally, so `march=native` wont harm anything.